Array of Pointer to Structure using Addresses Dereferencing by Index Failing to Parse

Ok, so I think this one is much simpler than the title may suggest, but it's the use-case I found this issue with, so I'm going to run with its full complexity for the sake of testing. Here's a snippet that seems to fail:

```pascal
FUNCTION_BLOCK class_SomethingCool
VAR
        currentChannel : ARRAY[1..g_c_someConstant] OF POINTER TO struct_groupData :=
                [       ADR(_object[1].someValue),
                        ADR(_object[2].someValue),
                        ADR(_object[3].someValue),
                        ADR(_object[4].someValue),
                        ADR(_object[5].someValue),
                        ADR(_object[6].someValue),
                        ADR(_object[7].someValue),
                        ADR(_object[8].someValue),
                        ADR(_object[9].someValue),
                        ADR(_object[10].someValue)
                ];
END_VAR
END_FUNCTION_BLOCK
```

With the following `pytest` exception (shortened significantly):

```python-traceback
E           lark.exceptions.UnexpectedCharacters: No terminal matches '[' in the current parser context, at line 4 col 36
E
E                           [       ADR(_object[1].someValue),
E                                              ^
E           Expected one of: 
E               * HASH
E               * LPAR
E               * COMMA
E               * RPAR
E               * ASSIGNMENT

C:\Python311\Lib\site-packages\lark\parsers\xearley.py:119: UnexpectedCharacters
```

Development branch is here: https://github.com/engineerjoe440/blark/tree/bugfix/advanced-array-development Clearly, I've added a simple sanity-check test, but haven't completed a fix, yet. May not have a chance until later this week at the earliest.

--- a/blark/iec.lark
+++ b/blark/iec.lark
@@ -339,7 +339,7 @@ array_initialization: "[" array_initial_element ( "," array_initial_element )* "
 array_initial_element: ( integer | enumerated_value ) "(" [ _array_initial_element ] ")" -> array_initial_element_count
                      | _array_initial_element
 
-_array_initial_element: constant
+_array_initial_element: expression
                       | structure_initialization
                       | enumerated_value
                       | array_initialization

--- a/blark/tests/source/array_of_addresses.st
+++ b/blark/tests/source/array_of_addresses.st
@@ -1,0 +1,16 @@
+FUNCTION_BLOCK class_SomethingCool
+VAR
+        currentChannel : ARRAY[1..g_c_someConstant] OF POINTER TO struct_groupData :=
+                [       ADR(_object[1].someValue),
+                        ADR(_object[2].someValue),
+                        ADR(_object[3].someValue),
+                        ADR(_object[4].someValue),
+                        ADR(_object[5].someValue),
+                        ADR(_object[6].someValue),
+                        ADR(_object[7].someValue),
+                        ADR(_object[8].someValue),
+                        ADR(_object[9].someValue),
+                        ADR(_object[10].someValue)
+                ];
+END_VAR
+END_FUNCTION_BLOCK

--- a/blark/transform.py
+++ b/blark/transform.py
@@ -3373,7 +3373,7 @@ Constant = Union[
 
 
 ArrayInitialElementType = Union[
-    Constant,
+    Expression,
     StructureInitialization,
     EnumeratedValue,
     ArrayInitialization,

